api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -160,7 +160,7 @@ public DbUser updateUserWithRetries(
     int objectLockingFailureCount = 0;
     int statementClosedCount = 0;
     while (true) {
-      dbUser = userModifier.apply(dbUser);
```

Comment:
I don't claim to understand the Hibernate magic which causes us to get into an unsynced state, but the pattern looks like this.

AFAIK the extra save() won't hurt, and it may prevent sync issues.

Reply:
.... actually this makes it impossible to save the Profile.  Forget it.